agent/engine/execcmd/manager_start_linux.go

```diff
@@ -135,35 +143,44 @@ func (m *manager) doStartAgent(ctx context.Context, client dockerapi.DockerClien
 	}
 	execRes, err := client.CreateContainerExec(ctx, containerId, execCfg, dockerclient.ContainerExecCreateTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [create]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 
 	seelog.Debugf("Task engine [%s]: created ExecCommandAgent for container: %s -> docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	err = client.StartContainerExec(ctx, execRes.ID, dockerclient.ContainerExecStartTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [pre-start]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: sent ExecCommandAgent start signal for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	inspect, err := client.InspectContainerExec(ctx, execRes.ID, dockerclient.ContainerExecInspectTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [docker inspect]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: inspect ExecCommandAgent for container: %s -> pid: %d, exitCode: %d, running:%v, err:%v",
 		task.Arn, containerId, inspect.Pid, inspect.ExitCode, inspect.Running, err)
 
 	if !inspect.Running { //TODO: [ecs-exec] retry only for certain exit codes?
-		return StartError{
+		execMD.StoppedReason = fmt.Sprintf("Execute Command Agent process exited with exit code: %d", inspect.ExitCode)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{
 			error:     fmt.Errorf("ExecCommandAgent could not start for container (exitCode=%d)", inspect.ExitCode),
 			retryable: true,
 		}
 	}
 	seelog.Infof("Task engine [%s]: started ExecCommandAgent for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
-	container.SetExecCommandAgentMetadata(&apicontainer.ExecCommandAgentMetadata{
-		PID:          strconv.Itoa(inspect.Pid),
-		DockerExecID: execRes.ID,
-		CMD:          execAgentCmd,
-	})
-	return nil
+	nowTime := time.Now()
+	execMD.PID = strconv.Itoa(inspect.Pid)
+	execMD.DockerExecID = execRes.ID
+	execMD.CMD = execAgentCmd
+	execMD.StartedAt = nowTime
```

Comment:
nit: You can use `time.Now()` directly here and get rid of the `nowTime` variable